File: app/src/main/java/com/an/trailers/utils/AnimUtils.java
package com.an.trailers.utils;

import android.view.animation.Animation;
import android.view.animation.AnimationSet;
import android.view.animation.DecelerateInterpolator;
import android.view.animation.TranslateAnimation;

public class AnimUtils {

    public static Animation createBgImageInAnimation(int fromX, int toX, int transitionDuration) {
        TranslateAnimation translate = new TranslateAnimation(fromX, toX, 0, 0);
        translate.setDuration(transitionDuration);

        AnimationSet set = new AnimationSet(true);
        set.setInterpolator(new DecelerateInterpolator());
        set.addAnimation(translate);
        return set;
    }

    public static Animation createBgImageOutAnimation(int fromX, int toX, int transitionDuration) {
        TranslateAnimation ta = new TranslateAnimation(fromX, toX, 0, 0);
        ta.setDuration(transitionDuration);
        ta.setInterpolator(new DecelerateInterpolator());
        return ta;
    }
}

File: app/src/main/java/com/an/trailers/utils/AnimUtils.kt
package com.an.trailers.utils

import android.view.animation.Animation
import android.view.animation.AnimationSet
import android.view.animation.DecelerateInterpolator
import android.view.animation.TranslateAnimation

object AnimUtils {

    fun createBgImageInAnimation(fromX: Int, toX: Int, transitionDuration: Int): Animation {
        val translate = TranslateAnimation(fromX.toFloat(), toX.toFloat(), 0f, 0f)
        translate.duration = transitionDuration.toLong()

        val set = AnimationSet(true)
        set.interpolator = DecelerateInterpolator()
        set.addAnimation(translate)
        return set
    }

    fun createBgImageOutAnimation(fromX: Int, toX: Int, transitionDuration: Int): Animation {
        val ta = TranslateAnimation(fromX.toFloat(), toX.toFloat(), 0f, 0f)
        ta.duration = transitionDuration.toLong()
        ta.interpolator = DecelerateInterpolator()
        return ta
    }
}
